pkgs/development/python-modules/diffimg/default.nix

```diff
@@ -21,7 +22,10 @@ buildPythonPackage rec {
   # fix offered to upstream https://github.com/nicolashahn/diffimg/pull/6
   postPatch = ''
     substituteInPlace diffimg/test.py \
-      --replace "from diff import diff" "from diffimg.diff import diff"
+      --replace-warn "from diff import diff" "from diffimg.diff import diff"
```

Comment:
why not `replace-fail` ?

Reply:
`replace-warn` seems non-blocking, which seems more robust
is there a particular reason you're implying `replace-fail` is better?

Reply:
tldr we want to know if something we asked for didn't happen


if `replace`  was done to fix a build, then if it fails in the future (eg can no longer apply), then CI would fail.

but if `replace` was there to fix something that build/tests didn't cover then we would never know of a regression.

if we tried to replace something and it didn't happen, surely we would like to know ? I am happy to hear examples where it makes sense to ignore failure (to apply a replace )

Reply:
i don't really care if it applies or not, i care if the build succeeds
if it doesn't apply and the build succeeds regardless, in this case, it's better than the build failing
ofc, that does leave a line to be cleaned up, but i do occasionally read the output of the build process

i think this case is a fair example of wanting a warning instead of a failure
it is a very quick build, and if the test being fixed fails to be fixed, it itself will fail promptly after that
if the test's fix doesn't apply and the build still succeeds, that's an accidental success
which i consider better than an accidental failure

yes, both cases are accidents that need to be fixed, but one doesn't block the things that depend on it :)

Reply:
I'll leave it up to you

but I see this same as 

```
cp a b/
```

if either a or b doesn't exist, we expect this to fail

Reply:
woeps, confused the test's `replace-warn` with the line you actually commented on
though my argument is the same, if it succeeds at importing `diff` without this fix, that's better than failing to apply the fix

(and to be clear, i'm only changing `replace` because it emits a warning about being deprecated)